packages/local-cli/util/loadMetroConfig.js

```diff
@@ -135,6 +132,29 @@ module.exports = async function load(
     defaultConfig
   );
 
+  /**
+   * Metro configuration requires the following two properties to be defined
+   * statically. They can't be a function, unline `serializer.getModulesRunBeforeMainModule`.
+   *
+   * We only resolve React Native files if no other value was provided by the user.
+   *
+   * This is to allow using CLI inside React Native source code, where there is no
+   * "react-native" module in "node_modules".
+   *
+   * See "react-native/metro.config.js"
+   */
+  if (!config.resolver.hasteImplModulePath) {
+    config.resolver.hasteImplModulePath = require.resolve(
+      'react-native/jest/hasteImpl'
+    );
+  }
+
+  if (!config.transformer.assetRegistryPath) {
```

Comment:
metro seems have a default assetRegistryPath, so this check seems not enough.
```js
if (!config.transformer.assetRegistryPath || config.transformer.assetRegistryPath === 'missing-asset-registry-path')
```

My env & use case:
1. current RN master code (https://github.com/facebook/react-native/commit/5957c8b370fd44fd0439461dee1e7d9eab806866)
2. react-native-local-cli@1.0.0-alpha.5
3. Apply this PR's patch with some minor change, that replace `require('react-native/foo')` to `require('../../../foo')`.
In case the path is `/path/to/react-native/node_modules/react-native-local-cli/util/loadMetroConfig.js` 
Patch file at https://gist.github.com/Kudo/a51f7992d6d1405330a3dcbb1d9e90fd
4. Running ./scripts/packager.sh and test with RNTester Android